src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -780,11 +780,24 @@ void enforceNotNull(SyntaxNode? syntaxOpt, LocalState state)
 
                 foreach (var parameter in this.MethodParameters)
                 {
-                    if (parameterHasBadState(parameter, state))
+                    var slot = GetOrCreateSlot(parameter);
+                    if (slot <= 0)
+                    {
+                        continue;
+                    }
+
+                    var annotations = parameter.FlowAnalysisAnnotations;
```

Comment:
This code was moved from `parameterHasBadState` below. However `GetFlowAnalysisAnnotations(Symbol)` here would just dispatch out to `ParameterSymbol.FlowAnalysisAnnotations`.

It could make sense to change this to `GetParameterAnnotations` in order to avoid attribute binding cycles. Since I can't come up with a crashing scenario for this off the top of my head I would like to punt changes to it to #47635